cli/src/runner.rs

```diff
@@ -0,0 +1,176 @@
+use std::io;
+use std::io::stderr;
+use std::ops::Deref;
+use std::path::PathBuf;
+
+use analyzer::diagnostic::Diagnostic;
+use analyzer::engine::Engine;
+use analyzer::importer::ASTImporter;
+use analyzer::name::Name;
+use analyzer::relations::SourceId;
+use analyzer::steps::typing::apply_types;
+use ast::group::Block;
+use ast::Expr;
+use compiler::compile;
+use context::source::{Source, SourceSegmentHolder};
+use lexer::lexer::lex;
+use parser::err::ParseReport;
+use parser::parse;
+
+use crate::cli::{display_exprs, display_tokens, execute, Configuration};
+use crate::disassemble::display_bytecode;
+use crate::render::diagnostic::render_diagnostic;
+use crate::render::parse_error::render_parse_error;
+use crate::render::SourcesCache;
+use crate::source_importer::FileSourceImporter;
+
+/// returns true if any error occurred
+pub(crate) fn run(source: PathBuf, working_dir: PathBuf, config: Configuration) -> bool {
```

Comment:
```suggestion
pub(crate) fn run(source: PathBuf, source_dir: PathBuf, config: Configuration) -> bool {
```
The `bool` is still inverted.